### ISSUE TITLE
Meta - Builds are failing in Server Mode

### ISSUE THREAD

**Issue by bblommers** [spulec/moto#3098]
Last successful build: https://travis-ci.org/github/spulec/moto/jobs/702363984
Failing build (example): https://travis-ci.org/github/spulec/moto/jobs/702608025

Failing test:
```
===================================================================
ERROR: test_cognitoidp.test_change_password
----------------------------------------------------------------------
Traceback (most recent call last):

  File "/home/travis/virtualenv/python2.7.17/lib/python2.7/site-packages/nose/case.py", line 197, in runTest
    self.test(*self.arg)
  File "/home/travis/build/spulec/moto/moto/core/models.py", line 98, in wrapper
    result = func(*args, **kwargs)
  File "/home/travis/build/spulec/moto/tests/test_cognitoidp/test_cognitoidp.py", line 1224, in test_change_password
    outputs = authentication_flow(conn)
  File "/home/travis/build/spulec/moto/tests/test_cognitoidp/test_cognitoidp.py", line 1169, in authentication_flow
    ChallengeResponses={"USERNAME": username, "NEW_PASSWORD": new_password},
  File "/home/travis/virtualenv/python2.7.17/lib/python2.7/site-packages/botocore/client.py", line 316, in _api_call
    return self._make_api_call(operation_name, kwargs)
  File "/home/travis/virtualenv/python2.7.17/lib/python2.7/site-packages/botocore/client.py", line 635, in _make_api_call
    raise error_class(parsed_response, operation_name)
ResourceNotFoundException: An error occurred (ResourceNotFoundException) when calling the RespondToAuthChallenge operation: b22f073c-53fb-43ef-971c-65c161cd5924
```

No changes have been made to our Cognito IDP code in between.

## Cause:

A new version of boto3 was released (1.14.11 -> 1.14.12), which depends on a new version of botocore (1.17.11 -> 1.7.12)
Botocore has made a change to the CognitoIDP RespondToAuthChallenge response format:
https://github.com/boto/botocore/commit/2300e3a87e2835c93d832ee9c8ec7433cfa7fe18#diff-104dd1e3491647c1ef3da008f40f2abdR1331

Will continue to investigate

**Comment by bblommers** [comment]
## Root cause

The `respond_to_auth_challenge`-method is executed in a different region then the previous steps (`admin_create_user`, `admin_initiate_auth`). This means that the `respond_to_auth_challenge`-method does not have access to the previously created session/user.

## Explanation
The change in `botocore` removes the authentication for the `respond_to_auth_challenge`-method. This, in turn, means that the request no longer has the Authentication-header, which is used to determine the region to execute the request in.
Without Auth-header, the request is executed in the default region (us-east-1).
The other requests, with Auth-header, are executed in the user-defined region (us-west-2).

**Comment by bblommers** [comment]
We solved the problem of unsigned requests (and not knowing which region to use) here: https://github.com/spulec/moto/blob/master/moto/server.py#L29
Which assumes we always want to use the default region (us-east-1)

This was acceptable for all other unsigned requests, but does not work for this specific use case - where we rely on data created in a non-default region.

One hacky solution to ensure we always have access to the region supplied by the client is to extend the UserAgent-header with `region/..`, using the `user_agent_extra` property in the [botocore Config](https://botocore.amazonaws.com/v1/documentation/api/latest/reference/config.html#botocore.config.Config) object. 

I've created #3099 as an POC.
